### DISCUSSION TITLE
What are the id generation rules for v9 directus_files?

### DISCUSSION THREAD

**Discussion by cdwmhcc** [directus/directus#4685] [Archive]
In the v8 version, the ID of directus_files is stored as `int(11)`, but it becomes `char(36)` in v9. I don’t know what the id generation rules are in v9. How can I migrate the file data in v8 to v9? ?

**Comment by cdwmhcc** [comment]
I used the rules of uuid v5 to convert the file `id` in v8 to uuid and `uploaded_by` to the user id (uuid) of v9. Now I have successfully imported the directus_files data of v8 into v9.

The current data can be displayed correctly. Due to the abnormal configuration of Alibaba OSS in v9, the pictures cannot be displayed normally. This is a problem in another issue.

**Reply by rijkvanzanten** [reply]
You're correct. The IDs for files / users and some other system tables are now UUIDv4's (so 32-char UUIDs).

I actually made an automated migration tool that automatically re-creates these IDs and relationships. Might be worth giving a try 🙂  https://github.com/directus-community/migration-tool

**Reply by cdwmhcc** [reply]
Wow, you did this job a long time ago. Why didn't I find it earlier, I'm so stupid!
